src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterFactory.cs

```diff
@@ -14,6 +14,8 @@ namespace System.Text.Json.Serialization
     /// </remarks>
     public abstract class JsonConverterFactory : JsonConverter
     {
+        internal const string JsonConverterFactoryRequiresDynamicCodeMessage = "Types deriving from JsonConverterFactory cannot be statically analyzed and might require runtime code generation.";
```

Comment:
One reason why the const string was moved here is that defining it inside `JsonStringEnumConverter` itself will prompt a self-referencial RDC warning trigger by the RDC attribute itself consuming a member of the annotated type.

Reply:
This message is only used in a single place. We do not need a string constant for it.